database/insertData.js

```diff
@@ -1690,6 +1950,75 @@ const queries = [
       }
     }
   }`,
+  `mutation ReviewTestApplication {
+    createApplication(
+      input: {
+        application: {
+          serial: "12345"
+          applicationSectionsUsingId: {
+            create: [{ templateSectionId: 6 }, { templateSectionId: 7 }]
+          }
+          name: "Test Review -- Vitamin C"
+          outcome: PENDING
+          templateId: 4
+          userId: 5
+          applicationStageHistoriesUsingId: {
+            create: {
+              isCurrent: true
+              templateStageToStageId: { connectById: { id: 5 } }
+              applicationStatusHistoriesUsingId: {
+                create: { isCurrent: true, status: SUBMITTED }
+              }
+            }
+          }
+          applicationResponsesUsingId: {
+            create: [
+              { isValid: null, value: null, templateElementId: 39 }
+              { isValid: true, templateElementId: 40, value: { text: "John" } }
+              { isValid: true, templateElementId: 41, value: { text: "Smith" } }
+              {
+                isValid: true
+                templateElementId: 42
+                value: { text: "js@nowhere.com" }
+              }
+              { isValid: null, value: null, templateElementId: 43 }
+              { isValid: true, templateElementId: 44, value: { text: "39" } }
+              {
+                isValid: true
+                templateElementId: 45
+                value: { text: "New Zealand" }
+              }
+              { isValid: null, value: null, templateElementId: 46 }
+              {
+                isValid: true
+                templateElementId: 47
+                value: { text: "Vitamin C" }
+              }
+              {
+                isValid: true
+                templateElementId: 48
+                value: { text: "Natural Product", optionIndex: 1 }
+              }
+              { isValid: null, value: null, templateElementId: 49 }
+              { isValid: true, templateElementId: 50, value: { text: "50mg" } }
+              { isValid: true, templateElementId: 51, value: { text: "100" } }
+              {
+                isValid: true
+                templateElementId: 52
+                value: { text: "Turning orange" }
+              }
```

Comment:
I was thinking maybe not using the templateElementId... If that can be easily replaced by the combination of templateId + elementCode it could be clearer. And maybe less dependent on the order of elements, thinking if we add other elements (even in another template) these `Id` would be changed.

Reply:
I agree, but I'm not sure how to do that. Even the other applications in this file use `templateElementToTemplateElementId: { connectById: { id: 12 } }`. It's not exactly the same, but it still refers to a specific templateElementID.

Any ideas? It'd be great if we could reference the template elements by code, I'm just sure how to.

Reply:
I'll have a look now see if I can find anything to replace that.

Reply:
I don't think it's possible without (at the very least) additional plugins. Because you'd need to provide a connection to a unique identifier. Which would need to be the combination of Section_id and code. And postgraphile doesn't provide "create" connections to unique fields anyway, only in queries. There may be a plugin to open up this functionality, but will have to look for that in another job.